internal/ipmi/ipmi.go

```diff
@@ -5,42 +5,79 @@ import (
 	"context"
 	"fmt"
 	"net"
+	"os"
 	"os/exec"
 	"strings"
 
+	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
 )
 
 // Ipmi holds the date for an ipmi connection
 type Ipmi struct {
-	Username string
-	Password string
-	Host     string
-	ipmitool string
+	Username    string
+	Password    string
+	Host        string
+	ipmitool    string
+	cipherSuite string
+	log         logr.Logger
+}
+
+// Option for setting optional Ipmi values
+type Option func(*Ipmi)
+
+func WithIpmitoolPath(path string) Option {
+	return func(i *Ipmi) {
+		i.ipmitool = path
+	}
+}
+
+func WithCipherSuite(cipherSuite string) Option {
+	return func(i *Ipmi) {
+		i.cipherSuite = cipherSuite
+	}
+}
+
+func WithLogger(log logr.Logger) Option {
+	return func(i *Ipmi) {
+		i.log = log
+	}
 }
 
 // New returns a new ipmi instance
-func New(username string, password string, host string) (ipmi *Ipmi, err error) {
+func New(username string, password string, host string, opts ...Option) (ipmi *Ipmi, err error) {
 	ipmi = &Ipmi{
 		Username: username,
 		Password: password,
 		Host:     host,
+		log:      logr.Discard(),
 	}
 
-	ipmi.ipmitool, err = exec.LookPath("ipmitool")
-	if err != nil {
-		return nil, err
+	for _, opt := range opts {
+		opt(ipmi)
+	}
+
+	if ipmi.ipmitool == "" {
+		ipmi.ipmitool, err = exec.LookPath("ipmitool")
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		fmt.Println("Using ipmitool path: ", ipmi.ipmitool)
+		if _, err = os.Stat(ipmi.ipmitool); err != nil {
```

Comment:
It's a difference in when the error will occur and it's in line with the existing `exec.LookPath` behavior. With this, the error occurs when `ipmi.New` is called instead of when `Open` or another method is called. This creates a different user experience. 
Walking this thought though, If we remove the `os.Stat` would we also remove the `exec.LookPath`?

Reply:
Oh right I see your point about the user experience, lets leave it as is then.